cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.operator.cluster.operator.resource;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.zjsonpatch.JsonDiff;
+import io.strimzi.operator.cluster.model.KafkaConfiguration;
+import io.strimzi.operator.cluster.model.KafkaVersion;
+import io.strimzi.operator.cluster.model.ModelUtils;
+import io.strimzi.operator.common.operator.resource.AbstractResourceDiff;
+import org.apache.kafka.clients.admin.AlterConfigOp;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static io.fabric8.kubernetes.client.internal.PatchUtils.patchMapper;
+
+/**
+ * Computes a diff between current config supplied as a Map ConfigResource, Config and desired config supplied as a ConfigMap
+ * An algorithm:
+ *  1. Create map from supplied desired ConfigMap
+ *  2. Fill placeholders (e.g. ${BROKER_ID}) in desired map
+ *  3a. If the entry is in IGNORABLE_PROPERTIES or entry.value from desired is equal to entry.value from current, do nothing
+ *      else add it to the diff
+ *  3b. If entry was removed from desired, add it to the diff with default value.
+ *  3c. If custom entry was removed, delete property
```

Comment:
What's a custom entry?

Reply:
If user adds to the `spec.kafka.config` any custom property.